src/components/Router/StepComponentMap.js

```diff
@@ -140,25 +143,50 @@ const getRequiredSelfieSteps = (deviceHasCameraSupport) => {
   return allSelfieSteps
 }
 
+const getNonPassportFrontDocumentCaptureFlow = (
+  hasPreselectedDocument,
+  showCountrySelection
+) => {
+  const frontCaptureComponents = [FrontDocumentCapture, DocumentFrontConfirm]
+  if (hasPreselectedDocument && showCountrySelection) {
+    return [CountrySelector, ...frontCaptureComponents]
+  }
+  if (hasPreselectedDocument && !showCountrySelection) {
+    return frontCaptureComponents
+  }
+  return [SelectIdentityDocument, CountrySelector, ...frontCaptureComponents]
+}
+
 const getIdentityDocumentComponents = (
   documentType,
   hasPreselectedDocument,
+  showCountrySelection,
   shouldUseCameraForDocumentCapture
 ) => {
-  const double_sided_docs = ['driving_licence', 'national_identity_card']
+  const doubleSidedDocs = ['driving_licence', 'national_identity_card']
+  const isPassportDocument = documentType === 'passport'
   const isDocumentUpload = !shouldUseCameraForDocumentCapture
-  const frontCaptureComponents =
-    documentType === 'passport' && isDocumentUpload
-      ? [FrontDocumentCapture, ImageQualityGuide, DocumentFrontConfirm]
-      : [FrontDocumentCapture, DocumentFrontConfirm]
-  const withSelectScreen =
-    documentType === 'passport'
-      ? [SelectIdentityDocument, ...frontCaptureComponents]
-      : [SelectIdentityDocument, CountrySelector, ...frontCaptureComponents]
-  const frontDocumentFlow = hasPreselectedDocument
-    ? frontCaptureComponents
-    : withSelectScreen
-  if (double_sided_docs.includes(documentType)) {
+
+  if (isPassportDocument) {
```

Comment:
The logic in this `getIdentityDocumentComponents` is getting complex. We should think of a way to refactor this (out of scope)

Reply:
Agreed. I tried to refactor it a bit here, but conscious of spending too much time on it.